From 2bf8612a61b61c1bbe65b64e3e659a0819aed099 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Tue, 17 Aug 2010 17:44:22 +0800
Subject: ENGR00126349-2 Change gfp flag in dma_alloc_coherent in common codes

Changed gfp flag directly in dma_alloc_coherent in
drivers/media/video/videobuf-dma-contig.c

Signed-off-by: Robby Cai <R63905@freescale.com>
(cherry picked from commit b5cfb266e2e2d40568d060b0426e6c4239a34a78)
---
 drivers/media/video/videobuf-dma-contig.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/video/videobuf-dma-contig.c b/drivers/media/video/videobuf-dma-contig.c
index 669f4ae..c47195b 100644
--- a/drivers/media/video/videobuf-dma-contig.c
+++ b/drivers/media/video/videobuf-dma-contig.c
@@ -284,7 +284,7 @@ static int __videobuf_mmap_mapper(struct videobuf_queue *q,
 
 	mem->size = PAGE_ALIGN(buf->bsize);
 	mem->vaddr = dma_alloc_coherent(q->dev, mem->size,
-					&mem->dma_handle, GFP_KERNEL);
+					&mem->dma_handle, GFP_DMA);
 	if (!mem->vaddr) {
 		dev_err(q->dev, "dma_alloc_coherent size %ld failed\n",
 			mem->size);
-- 
1.7.9.5

